api_tabular/utils.py

```diff
@@ -37,47 +106,90 @@ async def get_app_version() -> str:
-            # we're escaping the " because they are the encapsulators of the label
-            column = '"{}"'.format("__".join(column_split).replace('"', '\\"'))
-
-            if normalized_comparator == "sort":
-                if value == "asc":
-                    sql_query.append(f"order={column}.asc,__id.asc")
-                elif value == "desc":
-                    sql_query.append(f"order={column}.desc,__id.asc")
-                sorted = True
-            elif normalized_comparator == "exact":
-                sql_query.append(f"{column}=eq.{value}")
-            elif normalized_comparator == "differs":
-                sql_query.append(f"{column}=neq.{value}")
-            elif normalized_comparator == "contains":
-                sql_query.append(f"{column}=ilike.*{value}*")
-            elif normalized_comparator == "in":
-                sql_query.append(f"{column}=in.({value})")
-            elif normalized_comparator == "less":
-                sql_query.append(f"{column}=lte.{value}")
-            elif normalized_comparator == "greater":
-                sql_query.append(f"{column}=gte.{value}")
-            elif normalized_comparator == "strictly_less":
-                sql_query.append(f"{column}=lt.{value}")
-            elif normalized_comparator == "strictly_greater":
-                sql_query.append(f"{column}=gt.{value}")
+        _split = arg.split("=")
+        # filters are expected to have the syntax `<column_name>__<operator>=<value>`
+        if len(_split) == 2:
+            _filter, _sorted = add_filter(*_split)
+            if _filter:
+                sorted = sorted or _sorted
+                sql_query.append(_filter)
+        # aggregators are expected to have the syntax `<column_name>__<operator>`
+        elif len(_split) == 1:
+            column, operator = add_aggregator(_split[0])
+            if column:
+                aggregators[operator].append(column)
+        else:
+            raise ValueError
+    if aggregators:
+        agg_query = "select="
+        for operator in aggregators:
+            if operator == "groupby":
+                agg_query += f"{','.join(aggregators[operator])},"
+            else:
+                for column in aggregators[operator]:
+                    # aggregated columns are named `<column_name>__<operator>`
+                    # we pop the heading and trailing " that were added upstream
+                    # and put them around the new column name
+                    agg_query += f'"{column[1:-1]}__{operator}":{column}.{operator}(),'
+        # we pop the trailing comma (it's always there, by construction)
+        sql_query.append(agg_query[:-1])
     if page_size:
         sql_query.append(f"limit={page_size}")
     if offset >= 1:
         sql_query.append(f"offset={offset}")
-    if not sorted:
+    if not sorted and not aggregators:
         sql_query.append("order=__id.asc")
     return "&".join(sql_query)
 
 
+def get_column_and_operator(argument):
```

Comment:
Type hints on this method also would be great :)

Reply:
Oops thanks for noticing!